30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,19 @@
 Use grep to print only the lines from "input" for which:
 
 1. There is a binary number in the line
+	cat input |grep -a [0-9]
 
 2. There is an even binary number in the line
-
+	cat input |grep -a *[0-9]
 3. Line includes a hexadecimal number
-
+	cat input |grep '[xNN]'
 4. Line starts with a lowercase letter
-
+	cat input |grep ^[a-z]
 5. Line is not empty
-
+	cat input|grep -v -e '^$'
 6. Line starts with a lowercase letter and ends in a number
-
+	cat input |grep ^[a-z] $[0-9]
```

Comment:
A space in regular expression translates to a space, not to an AND. To create an AND you need to add another grep for example: `cat input | grep '^[a-z]'|grep '[0-9]$'`

Also, note $ must appear as the last character for it represents end-of-line